src/tableFormatter.ts

```diff
@@ -142,33 +155,58 @@ class MarkdownDocumentFormatter implements DocumentFormattingEditProvider {
         });
 
         // Normalize the num of hyphen, use Math.max to determine minimum length based on dash-line format
-        lines[1] = lines[1].map((cell, i) => {
+        lines[delimiterRowNum] = lines[delimiterRowNum].map((cell, i) => {
             if (/:-+:/.test(cell)) {
                 //:---:
-                colWidth[i] = Math.max(colWidth[i], 5);
-                colAlign[i] = 'c';
-                return ':' + '-'.repeat(colWidth[i] - 2) + ':';
+                const minimalWidth = delimiterRowNoPadding ? 5 : 5 + cellPaddingLength;
+                const numberOfColons = 2;
+
+                colWidth[i] = Math.max(colWidth[i], minimalWidth);
+                colAlign[i] = ColumnAlignment.Center;
+
+                let numberOfDelimiters = this.calculateNumberOfDelimiters(colWidth[i], numberOfColons, delimiterRowNoPadding, cellPaddingLength);
+
+                return ':' + '-'.repeat(numberOfDelimiters) + ':';
```

Comment:
(continued) So we don't have to introduce another `minimalWidth` and also `calculateNumberOfDelimiters` (just +2 if no padding).

Your current implementation is to add `cellPaddingLength` to `width` and `minimalWidth`, and then subtract it (conditionally) in `calcNumDelim`... We could have added it only when needed.

Reply:
See my reaction to your last comment why I did it this way.